cpp/ql/lib/semmle/code/cpp/internal/ResolveGlobalVariable.qll

```diff
@@ -0,0 +1,60 @@
+private predicate hasDefinition(@globalvariable g) {
+  exists(@var_decl vd | var_decls(vd, g, _, _, _) | var_def(vd))
+}
+
+pragma[noinline]
+private predicate onlyOneCompleteGlobalVariableExistsWithMangledName(@mangledname name) {
+  strictcount(@globalvariable g | hasDefinition(g) and mangled_name(g, name)) = 1
```

Comment:
I imagine this `strictcount` could be a `unique` (which might perform _slightly_ better).

Reply:
Yeah, indeed. Thanks for spotting that. I think the same holds for the one in `ResolveClass.qll` (I'll see if I can update that one in a separate PR).

Reply:
I assume you're expecting:
```codeql
  exists(unique(@globalvariable g | hasDefinition(g) and mangled_name(g, name)))
```
?

Reply:
Yeah, something like that

Reply:
Done. I'll re-do the DCA experiment just out of sheer paranoia.

Reply:
Reverted, after internal discussion about `unique` and `strictcount`.